Javascript-2/Week-1/Movie App/script.js

```diff
@@ -0,0 +1,201 @@
+import { movies } from "./arrayList.js";
+
+//////////////////////Streaming Function
+const nowStreaming = document.querySelector(".streaming");
+
+function createMovieCard(movie) {
+  //Creat a cotainer to store the Image and readMore Button
+
+  const movieContainer = document.createElement("div");
+  movieContainer.className = "movie-container";
+
+  //
+  //  Create an img element and set its src attribute
+  const img = document.createElement("img");
+  img.width = 200;
+  img.src = movie.poster_url;
+
+  // Create a button element
+  const readMoreBtn = document.createElement("button");
+  readMoreBtn.textContent = "About";
+  readMoreBtn.className = "accordion";
+
+  // starRating
+  const starRating = document.createElement("div");
+  starRating.className = "star-rating";
+  const numberOfStars = 5;
+  let stars = [];
+
+  for (let x = 0; x < numberOfStars; x++) {
+    const star = document.createElement("i");
+    star.className = "rating__star far fa-star";
+    starRating.appendChild(star);
+    stars.push(star);
+  }
+  const starClassActive = "rating__star fas fa-star";
+  const starClassInactive = "rating__star far fa-star";
+  stars.forEach((star, clickedStarIndex) => {
+    star.onclick = () => {
+      stars.forEach((star, starIndex) => {
+        if (starIndex <= clickedStarIndex) {
+          star.className = starClassActive;
+        } else {
+          star.className = starClassInactive;
+        }
+      });
+    };
+  });
+
+  movieContainer.appendChild(img);
+  movieContainer.appendChild(readMoreBtn);
+  movieContainer.appendChild(starRating);
+
+  // Create element to show details about the movie
+  const movieDetails = document.createElement("div");
+  movieDetails.className = "movie-details";
+  movieDetails.style.display = "none";
+  movieDetails.style.color = "white";
+  movieDetails.innerHTML = `
+             <p><strong>Title:</strong> ${movie.title}</p>
+             <p><strong>Description:</strong> ${movie.description}</p>
+             <p><strong>Release Year:</strong> ${movie.movie_year}</p>
+             <p><strong>Director:</strong> ${movie.director}</p>
+             <p><strong>Actors:</strong> ${movie.actors.join(", ")}</p>
+         `;
+  movieContainer.appendChild(movieDetails);
+
+  // Adding the comment Section
+
+  function createCommentSection() {
```

Comment:
Avoid declaring new functions inside another one. IMO `createCommentSection` can be defined outside of `createMovieCard`.